File: JS/commands/promote.js
module.exports = {
    name: 'promote',
    description: 'Promote Joshs Blog!',
    execute(msg, args) {
      const embed = {
        "title": "Joshs Blog",
        "description": "My personal blog about Software Development, Sports, Music, and any more of my hobbies!",
        "url": "https://joshglasson.com",
        "color": Math.floor(Math.random()*16777215),
        "timestamp": Date.now(),
        "thumbnail": {
          "url": "https://joshglasson.com/logo.png"
        },
        "image": {
          "url": "https://joshglasson.com/logo.png"
        },
        "author": {
          "name": "Josh Glasson",
          "url": "https://joshglasson.com",
          "icon_url": "https://joshglasson.com/logo.png"
        }
      };
      msg.channel.send("Check out my blog here: ", { embed });
    },
};

File: Ruby/src/modules/commands/promote.rb
require 'discordrb/webhooks'

module Bot
    module DiscordCommands
        module Promote extend Discordrb::Commands::CommandContainer
            command :promote do |event|
                event.channel.send_embed("Check out my blog here: ") do |embed|
                    embed.title = "Joshs Blog"
                    embed.colour = Random.bytes(3).unpack1('H*')
                    embed.url = "https://joshglasson.com"
                    embed.description = "My personal blog about Software Development, Sports, Music, and any more of my hobbies!"
                    embed.timestamp = Time.at(Time.now())
                    embed.image = Discordrb::Webhooks::EmbedImage.new(url: "https://joshglasson.com/logo.png")
                    embed.thumbnail = Discordrb::Webhooks::EmbedThumbnail.new(url: "https://joshglasson.com/logo.png")
                    embed.author = Discordrb::Webhooks::EmbedAuthor.new(name: "Josh Glasson", url: "https://joshglasson.com", icon_url: "https://joshglasson.com/logo.png")
                end
            end
        end
    end
end

